### ISSUE TITLE
[Question] Chunking keys within a server

### ISSUE THREAD

**Issue by Luo-Liang** [apache/incubator-mxnet#6565]
Hi! I have a question regarding accuracy of chunking a large key into a smaller key within a single server. I know MXNET_KVSTORE_BIGBOUND_ARRAY is used to chunk a key to different servers for processing.

Our implementation now chunks a physical key into a series of virtual keys, and these virtual keys are sent individually to one central server for optimization. The server only sees the virtual keys. We still use the vars for the original physical keys to synchronize access to the buffers. The virtual keys use buffers of the original physical keys (we chunk the buffer and assign different segments to the virtual keys).

We then send these virtual keys directly to optimize. For example:

update(self, index, weight, grad, state):

In here, index is a virtual key, instead of the original key.

Our problem is in a 0.3% accuracy drop with MNIST. In the first epoch, chunking vs not chunking's train accuracy starts to differs in 100-150 batches, but seems to be identical in the first 100 batches.  We do not know what may cause the problem, but it would be great if you can answer the following questions:

1. Is it fine to chunk keys and send each segments individually to server for optimization in general?
2. Can this be related to the optimizer? The optimizer is setup with states with physical keys, and it does not understand virtual keys, so it assigns default state for the keys it doesn't know?
3. Can this be related to update in per-key parameters? For example, learning rate for individual parameter (it only updates the learning rate for parameter it understands i.e. physical keys but not virtual keys)?

Many thanks!

**Comment by Luo-Liang** [comment]
After some experiments...
1, yes
2. yes
3. yes

:|

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!
